File: golang/2352.go
func equalPairs(grid [][]int) int {
  length := len(grid)
  m := make(map[[200]int]int)
  arr:=[200]int{}
  for i:=0; i<length; i++ {
     copy(arr[:], grid[i])
     m[arr]++
  }

  res := 0

  for i:=0; i<length; i++ {
      arr = [200]int{}
      for j:=0; j<length; j++ {
        arr[j] = grid[j][i]
      }
      if v, ok:=m[arr]; ok {
        res += v
      }
  }
  return res
}

File: rust/2352.rs
impl Solution {
    pub fn equal_pairs(grid: Vec<Vec<i32>>) -> i32 {
        use std::collections::HashMap;
        let mut res = 0i32;
        let mut m_r = HashMap::new();

        for v in grid.iter() {
            let s = v.into_iter().map(|n| n.to_string()).collect::<Vec<String>>().join(",");
            *m_r.entry(s).or_insert(0) += 1;
        }
        
        for i in 0..grid.len() {
            let mut col = vec![0; grid.len()];
            for j in 0..grid.len() {
                col[j] = grid[j][i];
            }
            let key = col.into_iter().map(|n|n.to_string()).collect::<Vec<String>>().join(",");

            if let Some(count) = m_r.get(&key) {
                res += count;
            }
        }
        res
    }
}
